### PR TITLE
Add istio-proxy and allow watching on different label

### PR DESCRIPTION
**Pull request by anderssonw**
Adds a new container type, and also allows watching different labels than nais' default.

Also makes the README somewhat more generic, and adds an explanation for the Label Selector.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -57,7 +57,8 @@ async fn main() -> anyhow::Result<()> {
             .unwrap();
     });
 
-    Controller::new(pods, watcher::Config::default().labels("nais.io/naisjob=true"))
+    let label_env = env::var("WATCH_SELECTOR").unwrap_or("nais.io/naisjob=true".to_string());
```
Comment (mortenlj) [inline src/main.rs:60]: I think I would prefer that this line happens earlier (almost at the start), just to keep "configuration" on more or less the same place.

**Review by mortenlj — CHANGES_REQUESTED** [review]
Just one minor complaint, otherwise it looks good.

File: src/main.rs (hunk 1 above)
Comment (anderssonw) [reply src/main.rs:60]: Fair! On it :)

**Review by mortenlj — APPROVED** [review]

**Comment by mortenlj** [comment]
If @Reasonable-Solutions gives his blessing, I think this can be merged :slightly_smiling_face:

**Review by Reasonable-Solutions — APPROVED** [review]

**Pull request merged**